[BUG] Disallowed call from unknown notification listener

### Describe the bug

```
Lawnchair bug report 10.08.2022
Lawnchair version: 9.1 Alpha 3 (9012942)
version.release: 11
version.sdk_int: 30

error: Uncaught exception

--------- beginning of stacktrace
java.lang.SecurityException: Disallowed call from unknown notification listener: android.service.notification.INotificationListener$Stub$Proxy@8629470 class android.service.notification.INotificationListener$Stub$Proxy
	at android.os.Parcel.createExceptionOrNull(Parcel.java:2376)
	at android.os.Parcel.createException(Parcel.java:2360)
	at android.os.Parcel.readException(Parcel.java:2343)
	at android.os.Parcel.readException(Parcel.java:2285)
	at android.app.INotificationManager$Stub$Proxy.getActiveNotificationsFromListener(INotificationManager.java:4806)
	at android.service.notification.NotificationListenerService.getActiveNotifications(NotificationListenerService.java:937)
	at android.service.notification.NotificationListenerService.getActiveNotifications(NotificationListenerService.java:914)
	at com.android.launcher3.notification.NotificationListener.onNotificationRankingUpdate(NotificationListener.java:298)
	at android.service.notification.NotificationListenerService$MyHandler.handleMessage(NotificationListenerService.java:2150)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:236)
	at android.app.ActivityThread.main(ActivityThread.java:8061)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:656)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:967)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.notification.ManagedServices.checkServiceTokenLocked(ManagedServices.java:904)
	at com.android.server.notification.NotificationManagerService$10.getActiveNotificationsFromListener(NotificationManagerService.java:4240)
	at android.app.INotificationManager$Stub.onTransact(INotificationManager.java:2182)
	at android.os.Binder.execTransactInternal(Binder.java:1162)
	at android.os.Binder.execTransact(Binder.java:1126)


```

### Steps to reproduce

Steps to reproduce the behavior:
1. Go to '...' lawnvhair
2. Click on '....'
3. Scroll down to '....'
4. See error


### Expected behavior

Don't throw exceptions 

### Screenshots

_No response_

### Device information

- Device: Android Smartphone
- OS:  Android version 11
- App version  9.1 Alpha 3 (9012942)

### Additional context

_No response_

--- a/src/com/android/launcher3/notification/NotificationListener.java
+++ b/src/com/android/launcher3/notification/NotificationListener.java
@@ -141,7 +141,7 @@ public class NotificationListener extends NotificationListenerService {
                 if (notificationGroup != null) {
                     notificationGroup.removeChildKey(key);
                     if (notificationGroup.isEmpty()) {
-                        if (key.equals(mLastKeyDismissedByLauncher)) {
+                        if (key.equals(mLastKeyDismissedByLauncher) && sIsConnected) {
                             // Only cancel the group notification if launcher dismissed the
                             // last child.
                             cancelNotification(notificationGroup.getGroupSummaryKey());
@@ -149,7 +149,7 @@ public class NotificationListener extends NotificationListenerService {
                         mNotificationGroupMap.remove(sbn.getGroupKey());
                     }
                 }
-                if (key.equals(mLastKeyDismissedByLauncher)) {
+                if (key.equals(mLastKeyDismissedByLauncher) && sIsConnected) {
                     mLastKeyDismissedByLauncher = null;
                 }
                 return true;
@@ -172,8 +172,10 @@ public class NotificationListener extends NotificationListenerService {
                 mUiHandler.obtainMessage(message.what, activeNotifications).sendToTarget();
                 return true;
             case MSG_CANCEL_NOTIFICATION: {
-                mLastKeyDismissedByLauncher = (String) message.obj;
-                cancelNotification(mLastKeyDismissedByLauncher);
+                if (sIsConnected) {
+                    mLastKeyDismissedByLauncher = (String) message.obj;
+                    cancelNotification(mLastKeyDismissedByLauncher);
+                }
                 return true;
             }
             case MSG_RANKING_UPDATE: {

